src/vector_transport.jl

```diff
@@ -88,6 +138,37 @@ struct PoleLadderTransport{
     end
 end
 
+@doc raw"""
+    ScaledVectorTransport{T} <: AbstractVectorTransportMethod
+
+Introduce a scaled variant of any [`AbstractVectorTransportMethod`](@ref) `T`,
+as introduced in [^SatoIwai2013] for some ``X\inT_p\mathcal M`` as
+
+```math
+    \mathcal T^{\mathrm{S}}(X) = \frac{\lVert X\rVert_p}{\lVert \mathcal T(x)\rVert_q}\mathcal T(x).
+```
+
+Note that the resulting point `q` has to be known, i.e. for [`vector_transport_direction`](@ref)
+the curve or more precisely its end point has to be known (via an exponential map or a
+retraction). Therefore a default implementation is only provided for the [`vector_transport_to`](@ref)
+
+# Constructor
+
+    ScaledVectorTransport(m::AbstractVectorTransportMethod)
+
+[^SatoIwai2013]:
+    > Sato, H., Iwai, T.: _A new, globally convergent Riemannian conjugate gradient method_,
+    > Optimization, 2013, Volume 64(4), pp. 1011–1031.
+    > doi: [10.1080/02331934.2013.836650](https://doi.org/10.1080/02331934.2013.836650),
+    > arXiv: [1302.0125](https://arxiv.org/abs/1302.0125).
+"""
+struct ScaledVectorTransport{T<:AbstractVectorTransportMethod} <:
```

Comment:
That paper by Sato states that scaled vector transport is not a vector transport because it's not linear. Are our `AbstractVectorTransportMethod`s supposed to be linear or not?

Reply:
uff I don't know, I initially just thought those are easy to add and now I am for ages in fixing ambiguities again. Maybe its better to leave it out then?

Reply:
I can solve these ambiguities :slightly_smiling_face:

Reply:
But you are right, we usually assume the vector transport to be linear and all that I know are, so the question is then, how to handle this one...

Reply:
Ah it seems the ambiguities are resolved now (still I work ages on those); so the question is, do we want to have it though it is nonlinear?

Reply:
It appears that there are no ambiguities now though? The one on nightly is unrelated.

Reply:
Well, it took me 30 minutes of pondering around to find and fix 2 or 3 of them.

Reply:
In case there are some applications that require a linear vector transport, we can also have a `AbstractLinearVectorTransport` type.

Reply:
But that would require specifying most of our already-existing ones to declare them linear, right?

Reply:
Yes, of course. I can do that if you want.

Reply:
Let me just shortly finish show & tests.